### ISSUE TITLE
Optimise timeout timers

### ISSUE THREAD

**Issue by stefangotz** [echronos/echronos#33]
# Motivation

Some RTOS APIs block the caller up until a specified timeout, for example `message_queue_wait_timeout()`, `mutex_lock_timeout()`, and `sem_wait_timeout()`.
Currently, the implementation of this feature may disable and re-request a timer object multiple times per API call.
Technically, it should be possible to prime the internal timer just once per API call.
This approach would be more efficient in code space and run time.

# Goals

The goal of this task is to assess the feasibility of priming the internal timer of `..._timeout()` APIs only once per call.
Overly tight coupling between components or variants shall be avoided.
If that is feasible, the improvement shall be implemented.

# Test Plan

- code review
- system tests shall show that system behavior is not affected by this change

**Comment by stefangotz** [comment]
Duplicate of #7
